File: LeetCode/153.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1
        res = nums[0]

        if nums[0] < nums[-1]:
            return res

        while left <= right:

            mid = (left + right) // 2
            res = min(res, nums[mid])

            if nums[mid] < nums[0]:
                right = mid - 1

            else:
                left = mid + 1

        return res

File: leetcode-new/153.cpp
class Solution
{
public:
    int findMin(vector<int> &nums)
    {
        int l = 0;
        int r = nums.size() - 1;
        while (l < r - 1)
        {
            int m = (l + r) >> 1;
            if (nums[m] < nums[l])
                r = m;
            else
                l = m;
        }
        return min(nums[0], nums[r]);
    }
};
